### PR TITLE
[15.0][MIG] account_spread_cost_revenue

### PR DESCRIPTION
**Pull request by Saran440**
This module migration from standard and add PR enhance https://github.com/OCA/account-financial-tools/pull/1580

### CODE REVIEW

File: account_spread_cost_revenue/models/account_move_line.py [hunk 1]
```diff
@@ -97,7 +97,9 @@ def write(self, vals):
     def _check_spread_reconcile_validity(self):
         # Improve error messages of standard Odoo
         reconciled_lines = self.filtered(lambda l: l.reconciled)
-        msg_line = _("Move line: %s (%s), account code: %s\n")
+        msg_line = _(
+            "Move line: %(line_id)s (%(line_name)s), account code: %(account_code)s\n"
```
Comment (astirpe) [inline account_spread_cost_revenue/models/account_move_line.py:101]: Why changing this line? This causes a failure when executing this line:  `msg += msg_line % (line.id, line.name, line.account_id.code)`

Comment (BT-anieto) [reply account_spread_cost_revenue/models/account_move_line.py:101]: Agree with @astirpe. Could you check it please @Saran440 ?

File: account_spread_cost_revenue/models/account_move_line.py [hunk 2]
```diff
@@ -146,10 +148,9 @@ def _filter_line(aline, iline):
             elif len(template) > 1:
                 raise UserError(
                     _(
-                        "Too many auto spread templates (%s) matched with the "
-                        "invoice line, %s"
-                    )
-                    % (len(template), line.display_name)
+                        "Too many auto spread templates ({}) matched with the "
+                        "invoice line, {}"
+                    ).format(len(template), line.display_name)
```
Comment (BT-anieto) [inline account_spread_cost_revenue/models/account_move_line.py]: Do not change the translations. Format is not recommended.

File: account_spread_cost_revenue/models/account_spread.py [hunk 3]
```diff
@@ -520,7 +520,7 @@ def _message_post_unlink_invoice_line(self):
                 "<a href=# data-oe-model=account.move "
                 "data-oe-id=%d>%s</a>" % (spread.invoice_id.id, _("Invoice"))
             )
-            msg_body = _("Unlinked invoice line '%s' (view %s).") % (
+            msg_body = _("Unlinked invoice line '{}' (view {}).").format(
```
Comment (BT-anieto) [inline account_spread_cost_revenue/models/account_spread.py]: Do not change the translations. Format is not recommended

File: account_spread_cost_revenue/models/account_spread.py [hunk 4]
```diff
@@ -529,7 +529,7 @@ def _message_post_unlink_invoice_line(self):
                 "<a href=# data-oe-model=account.spread "
                 "data-oe-id=%d>%s</a>" % (spread.id, _("Spread"))
             )
-            msg_body = _("Unlinked '%s' (invoice line %s).") % (
+            msg_body = _("Unlinked '{}' (invoice line {}).").format(
```
Comment (BT-anieto) [inline account_spread_cost_revenue/models/account_spread.py]: Do not change the translations. Format is not recommended

File: account_spread_cost_revenue/models/account_spread_template.py [hunk 5]
```diff
@@ -169,7 +169,9 @@ def _check_auto_spread_ids_unique(self):
             )
         if results:
             raise UserError(
-                _("Followings are duplicated combinations,\n\n%s" % "\n".join(results))
+                _("Followings are duplicated combinations,\n\n{}").format(
```
Comment (BT-anieto) [inline account_spread_cost_revenue/models/account_spread_template.py]: Do not change the translations. Format is not recommended

**Review by BT-anieto — CHANGES_REQUESTED** [review]
Thx for the PR! Try to reduce the changes at maximum when it is a migration. Moreover, format is not recommended for odoo translations. Could you chan ge it?

On the other hand, ensure that you squash the commits following https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by Saran440** [comment]
@BT-anieto @astirpe Hi, Thank you for your review
I change it back to %(name)s because %s not support pre-commit it will error

```
account_spread_cost_revenue/models/account_move_line.py:100: [W8120(translation-positional-used), AccountMoveLine._check_spread_reconcile_validity] Translation method _('Move line: %s (%s), account code: %s\n') is using positional string printf formatting. Use named placeholder `_("%(placeholder)s")` instead.

account_spread_cost_revenue/models/account_move_line.py:148: [W8120(translation-positional-used), AccountMoveLine.create_auto_spread] Translation method _('Too many auto spread templates (%s) matched with the invoice line, %s') is using positional string printf formatting. Use named placeholder `_("%(placeholder)s")` instead.

account_spread_cost_revenue/models/account_spread.py:523: [W8120(translation-positional-used), AccountSpread._message_post_unlink_invoice_line] Translation method _("Unlinked invoice line '%s' (view %s).") is using positional string printf formatting. Use named placeholder `_("%(placeholder)s")` instead.

account_spread_cost_revenue/models/account_spread.py:532: [W8120(translation-positional-used), AccountSpread._message_post_unlink_invoice_line] Translation method _("Unlinked '%s' (invoice line %s).") is using positional string printf formatting. Use named placeholder `_("%(placeholder)s")` instead.

```

If you ok for this commit 33be08740bc71c36fdf420dc8ab312c2c749c788 i will merge this commit into 1 commit
What do you think?

**Comment by BT-anieto** [comment]
Hi @Saran440 , you still need to follow https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests
You need to squash the commits and you can only push 2 commits: one of the refactor and another one with the code changes.

**Comment by Saran440** [comment]
I found a bug in module account_spread_cost_revenue https://github.com/OCA/account-financial-tools/pull/1646/commits/e317f12a366268be8a8ad9b3c7c627663acb8256
Should we fixed it in this PR or next PR?

Step to error:
1. create spread templates
Selection_001

2. create vendor bills and select the same account as the template > confirm
3. Go to 'Spread Costs/Revenues' > select spread cost that create from template > click 'create move' on line.
Selection_002
4. Error because It creates a new move and **a new template** (i test by comment error. it will create new template and move_type is entry)

**Comment by astirpe** [comment]
@Saran440 I just ported your latest fix to V16 a6dae3b but with context passed in `def create_move()`. It's better to pass the context within `def create_move()` so that it also fixes the "Create All Moves" button (that calls the same method).

**Comment by Saran440** [comment]
@astirpe Fixed. I change it to pass context in function `def create_move()`. Thank you for reviewed.

**Review by ps-tubtim — APPROVED** [review]
Functional Test: Ok

**Review by astirpe — APPROVED** [review]

**Pull request merged**